src/pages/ServiceGraph/ServiceGraphPage.tsx

```diff
@@ -1,19 +1,72 @@
 import * as React from 'react';
+import * as API from '../../services/Api';
 import { RouteComponentProps } from 'react-router-dom';
+import PropTypes from 'prop-types';
 import NamespaceId from '../../types/NamespaceId';
 import CytoscapeLayout from '../../components/CytoscapeLayout/CytoscapeLayout';
+import { DropdownButton, MenuItem } from 'patternfly-react';
 
-export default function ServiceGraphPage(routeProps: RouteComponentProps<NamespaceId>) {
-  const namespace = routeProps.match.params.namespace;
+type ServiceGraphPageProps = {
+  availableNamespaces: { name: string }[];
+  graphNamespace: string;
+};
 
-  return (
-    <div className="container-fluid container-pf-nav-pf-vertical">
-      <div className="page-header">
-        <h2>Services Graph for namespace: {namespace} </h2>
-      </div>
-      <div style={{ height: 600 }}>
-        <CytoscapeLayout namespace={namespace} />
+export default class ServiceGraphPage extends React.Component<RouteComponentProps<NamespaceId>, ServiceGraphPageProps> {
+  static contextTypes = {
+    router: PropTypes.object
+  };
+
+  constructor(routeProps: RouteComponentProps<NamespaceId>) {
+    super(routeProps);
+    this.namespaceSelected = this.namespaceSelected.bind(this);
+
+    this.state = {
+      availableNamespaces: [],
+      graphNamespace: routeProps.match.params.namespace
+    };
+
+    this.populateNamespacesSelect = this.populateNamespacesSelect.bind(this);
+  }
+
+  componentDidMount() {
+    API.GetNamespaces().then(this.populateNamespacesSelect);
+  }
+
+  componentWillReceiveProps(nextProps: RouteComponentProps<NamespaceId>) {
+    this.setState({ graphNamespace: nextProps.match.params.namespace });
+  }
+
+  populateNamespacesSelect(response: any) {
+    this.setState({ availableNamespaces: response['data'] });
+  }
+
+  namespaceSelected(selectedNamespace: string) {
+    this.context.router.history.push(`/service-graph/${selectedNamespace}`);
+  }
+
+  render() {
+    return (
+      <div className="container-fluid container-pf-nav-pf-vertical">
+        <div className="page-header">
+          <h2>Services Graph for namespace:&nbsp;
+            <DropdownButton
+              id="namespace-selector"
+              title={this.state.graphNamespace}
+              onSelect={this.namespaceSelected}
+              bsSize="large"
+            >
+              {this.state.availableNamespaces.map(ns => (
+                <MenuItem key={ns.name} active={ns.name === this.state.graphNamespace} eventKey={ns.name}>
+                  {ns.name}
+                </MenuItem>
+              ))}
+            </DropdownButton>
+          </h2>
+        </div>
+        <div style={{ height: 600 }}>
```

Comment:
This breaks the stuff @cfcosta just added with the adjustable layout.

Reply:
I also had a doubt about this line. It's in the master branch, so I leave it there. But I can remove it as part of this PR if you want.

Reply:
It's removed.